packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -482,35 +496,43 @@ export class TokenDetectionController extends StaticIntervalPollingController<
       return;
     }
 
-    const addressAgainstWhichToDetect =
+    this.#addressAgainstWhichToDetect =
       selectedAddress ?? this.#selectedAddress;
-    const {
-      chainId: chainIdAgainstWhichToDetect,
-      networkClientId: networkClientIdAgainstWhichToDetect,
-    } = this.#getCorrectChainIdAndNetworkClientId(networkClientId);
+    const { chainId, networkClientId: selectedNetworkClientId } =
+      this.#getCorrectChainIdAndNetworkClientId(networkClientId);
+    this.#chainIdAgainstWhichToDetect = chainId;
+    this.#networkClientIdAgainstWhichToDetect = selectedNetworkClientId;
 
-    if (!isTokenDetectionSupportedForNetwork(chainIdAgainstWhichToDetect)) {
+    if (
+      !isTokenDetectionSupportedForNetwork(this.#chainIdAgainstWhichToDetect)
+    ) {
       return;
     }
-
     if (
       !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect !== ChainId.mainnet
+      this.#chainIdAgainstWhichToDetect !== ChainId.mainnet
     ) {
       return;
     }
     const isTokenDetectionInactiveInMainnet =
       !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect === ChainId.mainnet;
+      this.#chainIdAgainstWhichToDetect === ChainId.mainnet;
     const { tokensChainsCache } = this.messagingSystem.call(
       'TokenListController:getState',
     );
-    const tokenList =
-      tokensChainsCache[chainIdAgainstWhichToDetect]?.data ?? {};
-    const tokenListUsed = isTokenDetectionInactiveInMainnet
+    this.#tokenList = isTokenDetectionInactiveInMainnet
       ? STATIC_MAINNET_TOKEN_LIST
-      : tokenList;
+      : tokensChainsCache[this.#chainIdAgainstWhichToDetect]?.data ?? {};
 
+    for (const tokensSlice of this.#getSlicesOfTokensToDetect()) {
+      if (tokensSlice.length === 0) {
+        break;
+      }
```

Comment:
With the new code to slice dynamically instead of hardcoding length 2, this should no longer be needed / possible.  Although I don't mind it too much if you want to keep.

Reply:
Good catch. Don't see a reason to keep it. Removed here: https://github.com/MetaMask/core/pull/3938/commits/87b8f6aba4f5d057f40b25c6da4ee49dfb4a5a88